core/src/test/java/org/apache/calcite/test/SqlValidatorTestCase.java

```diff
@@ -64,6 +65,13 @@
   private static final SqlTestFactory EXTENDED_TEST_FACTORY =
       SqlTestFactory.INSTANCE.withCatalogReader(MockCatalogReaderExtended::new);
 
+  private static final SqlTestFactory NAMED_PARAM_TEST_FACTORY =
+      SqlTestFactory.INSTANCE.with("namedParamTable", "BodoNamedParams").withCatalogReader(
+          MockCatalogReaderSimpleNamedParam::new);
```

Comment:
I assume ::new calls init, which adds the tables to the CatalogReader?

Reply:
Yes.